SerialPrograms/Source/Integrations/DppIntegration/DppUtility.cpp

```diff
@@ -0,0 +1,87 @@
+#ifdef PA_DPP
+
+#include <dpp/dpp.h>
+#include <Integrations/DppIntegration/DppUtility.h>
+#include "CommonFramework/GlobalSettingsPanel.h"
+
+using namespace dpp;
+namespace PokemonAutomation {
+    namespace Integration {
+        namespace DppUtility {
+            std::map<std::string, std::atomic_int> Utility::user_counts;
+
+            Logger& Utility::dpp_logger() {
+                static TaggedLogger logger(global_logger_raw(), "DPP");
+                return logger;
+            }
+
+            void Utility::log(const std::string& message, const std::string& identity, const loglevel& ll) {
+                std::string log = identity + (std::string)": " + message;
+                Color color;
+                switch (ll) {
+                    case ll_debug: color = COLOR_CYAN; break;
+                    case ll_error: color = COLOR_RED; break;
+                    case ll_critical: color = COLOR_MAGENTA; break;
+                    default: color = COLOR_PURPLE; break;
+                };
+
+                dpp_logger().log(log, color);
+            }
+
+            void Utility::get_user_counts(cluster& bot, const guild_create_t& event) {
+                snowflake last = 0;
+                uint64_t count = 0;
+
+                // Retrieve ID and exit early if we have already pulled members for this guild.
+                auto id = std::to_string(event.created->id);
+                if (!user_counts.empty() && user_counts.count(id)) {
+                    log("Users for " + event.created->name + " already initialized.", "get_user_counts()", ll_info);
+                    return;
+                }
+
+                // Fetch the first 1000 members.
+                auto map = bot.guild_get_members_sync(id, 1000, last);
+                count += map.size();
+
+                /*Loop until we get fewer than 1000 because we've reached the end at that point.
+                  Determine the largest member ID for the next iteration. */
+                while (map.size() == 1000) {
+                    auto usr = std::max_element(map.begin(), map.end(), [](const auto& x, const auto& y) {
+                        return x.second.user_id < y.second.user_id;
+                    });
+
+                    last = usr->second.user_id;
+                    map = bot.guild_get_members_sync(id, 1000, last);
+                    count += map.size();
+                }
+
+                user_counts.emplace(id, count);
+                log("User count: " + std::to_string(count) + " (" + event.created->name + ")", "get_user_counts()", ll_info);
+            }
+
+            uint64_t Utility::sanitize_console_id_input(const slashcommand_t& event) {
+                int64_t id = std::get<int64_t>(event.get_parameter("id"));
+                if (id > UINT64_MAX) {
```

Comment:
I'm surprised you're not getting a compiler warning here about comparing signed and unsigned integer.

This if-statement will never return true since `id` will be cast to `uint64_t` first. Then no `uint64_t` can be greater than `UINT64_MAX`.

Are you trying to check for negative `id`?

Reply:
I'm actually not sure why I still have this method here, I had it before I learned I can enforce min values in slash commands, and then it did have a negative check. Will yeet in just a moment.